shell/platform/common/cpp/text_input_model.cc

```diff
@@ -40,24 +45,100 @@ void TextInputModel::SetText(const std::string& text) {
   text_ = utf16_converter.from_bytes(text);
   selection_base_ = 0;
   selection_extent_ = 0;
+  composing_base_ = 0;
+  composing_extent_ = 0;
```

Comment:
I think we should make a simple range class to contain these, the `InRange` utility (maybe called `Contains(...)`), an `IsEmpty()` since there are a bunch of checks conceptually doing that, `start()`/`end()` instead of having two copies for the two ranges, etc.

Reply:
Was thinking about this last night before I went to sleep, actually. Completely agreed. Will do.

Reply:
Done in #21722.